doc/maintaining_repo.md

```diff
@@ -6,22 +6,22 @@ The Dockerfiles used in Offical Docker Hub repos for OSRF projects are hosted he
+ * The way project Debian packages are installed must include the specific package version number. This is to ensure that the version to be installed is explicitly stated and will not happen to change over the course of the domino effect of automated build stages. This also helps keep builds repeatable and reproducible, and is a good practice to follow when making your own tags from the official images.
 
 ## Update Procedure
-The officla Docker Hub repo are required to be kept up to date. Security fix on an OS level usually come upstream, for us this is from the Offical Ubuntu image. However for our automatic rebuild triggered from ubuntu to work, the Debian packages referenced in the Dockerfile need to be installable.
+The official Docker Hub repo are required to be kept up to date. Security fix on an OS level usually come upstream, for us this is from the Offical Ubuntu image. However for our automatic rebuild triggered from Ubuntu to work, the Debian packages referenced in the Dockerfile need to be installable.
 
 ### ROS
-> One caviot for ROS's Debian package disrobutoin system is that [reprepro](http://mirrorer.alioth.debian.org/) can only host one version of a package ([as in FAQ 3.1](http://anonscm.debian.org/gitweb/?p=mirrorer/reprepro.git;a=blob_plain;hb=HEAD;f=docs/FAQ)), i.e. the latest one. In this case, the Dockerfile for the offical ROS repo should be updated accordingly, as the build of the Dockerfile will fial when apt-get fails to download the older package versions.
+> One caveat for ROS's Debian package distribution system is that [reprepro](http://mirrorer.alioth.debian.org/) can only host one version of a package ([as in FAQ 3.1](http://anonscm.debian.org/gitweb/?p=mirrorer/reprepro.git;a=blob_plain;hb=HEAD;f=docs/FAQ)), i.e. the latest one. In this case, the Dockerfile for the offical ROS repo should be updated accordingly, as the build of the Dockerfile will fail when apt-get fails to download the older package versions.
 
-> This can be done simply however by using the scripts described in accompanying docs to regenerate all Dockerfiles. For ROS, the scripts will use ros_buildfarm.common package to fetch the latest package index and append the proper version numbers.
+> This can be done simply however by using the scripts described in accompanying docs to regenerate all Dockerfiles. For ROS, the scripts will use `ros_buildfarm.common` package to fetch the latest package index and append the proper version numbers.
 
-Also, we now pin the version used for each ROS package. Note that however, we are using reprepro (a tool to handle local repositories of debian packages), and as in FAQ 3.1 is limited to one version per architectur. So from the Repeatability documentation referenced
+Also, we now pin the version used for each ROS package. Note that however, we are using reprepro (a tool to handle local repositories of debian packages), and as in FAQ 3.1 is limited to one version per architecture.
```

Comment:
@ruffsl Not sure what the last sentence was trying to communicate (so I removed it). Feel free to push directly to this branch if you think a sentence should be added back

Reply:
Looks like a trailing sentence I forgot about.

I probably meant to say something along the line of repeatability like with what I commented/linked in  https://github.com/osrf/docker_images/issues/171#issuecomment-404290452 .